api/src/main/java/net/automatalib/automata/visualization/MMCVisualizationHelper.java

```diff
@@ -0,0 +1,54 @@
+package net.automatalib.automata.visualization;
+
+import java.util.Map;
+
+import net.automatalib.automata.graphs.TransitionEdge;
+import net.automatalib.ts.modal.transitions.GroupMemberEdge;
+import net.automatalib.ts.modal.ModalContract;
+import net.automatalib.ts.modal.transitions.ModalContractEdgeProperty;
+
+public class MMCVisualizationHelper<S, I, T, TP extends ModalContractEdgeProperty & GroupMemberEdge, M extends ModalContract<S, I, T, TP>>
+        extends MCVisualizationHelper<S, I, T, TP, M> {
+
+    public MMCVisualizationHelper(M mc) {
+        super(mc);
+    }
+
+    @Override
+    public boolean getEdgeProperties(S src, TransitionEdge<I, T> edge, S tgt, Map<String, String> properties) {
+        if (!super.getEdgeProperties(src, edge, tgt, properties)) {
+            return false;
+        }
+
+        TP transitionProperty = super.automaton.getTransitionProperty(edge.getTransition());
+
+        int memberId = transitionProperty.getMemberId();
+        ModalContractEdgeProperty.EdgeColor color = transitionProperty.getColor();
+
+        if (color == ModalContractEdgeProperty.EdgeColor.GREEN) {
+            // use colors from 3 to 7 of colorscheme greens9
+            properties.put("colorscheme", "greens9");
+            properties.put(EdgeAttrs.COLOR, String.valueOf(7 - (memberId % 5)));
+
+        }
+        else if (color == ModalContractEdgeProperty.EdgeColor.RED) {
+            if (0 <= memberId && memberId <= 6) {
+                // use colors from 3 to 9 of colorscheme reds9
+                properties.put("colorscheme", "reds9");
+                properties.put(EdgeAttrs.COLOR, String.valueOf(9 - memberId));
+            }
+            else if (7 <= memberId && memberId <= 13) {
+                properties.put("colorscheme", "ylorrd9");
+                properties.put(EdgeAttrs.COLOR, String.valueOf(16 - memberId));
+            }
+            else if (14 <= memberId) {
+                properties.put("colorscheme", "rdpu9");
+                properties.put(EdgeAttrs.COLOR, String.valueOf(9 - (memberId % 7)));
+            }
+        }
```

Comment:
What purpose does the memberId have? Is it just for clustering? I think having it as a semantic property is fine, but things like `colorscheme` seems very DOT-specific (whereas the `VisualizationHelper` is intended to be visualization-agnostic) and magic numbers are a code smell. This kind of specific visualization would probably better fit in you application code.

Reply:
It is, indeed, a form of clustering. However, in our use case these ids are an intrinsic property of the edges. It is important that the mapping edge -> id is available and update when using this graph, esp. in contexts like subgraphs or copies. Therefore it seemed appropriate to store this information inside the graph itself.
As for the colorschemes, we use them to visualize the clusters depending on the memberId and the color. The magic numbers are used to ensure visually distinguishable colors. When the `VisualizationHelper` is not suitable for this tasks, can you extract the code? Or should it be client specific?

Reply:
Using `MMCAttrs.MEMBERSHIP` to associate an edge with a cluster is fine with me, but I think (DOT-specific) color markup should be client-specific (e.g. only *you* know that there are 14 clusters, etc.)

Reply:
Ok